Align mask formatting of both masks more closely

mikekgfb opened the pull request:
Summary: Align mask formatting of both masks more closely

Test Plan: sandcastle & github

Differential Revision: D43878634

cpuhrsch commented on test/test_transformers.py:
```diff
@@ -539,6 +539,45 @@ def test_padding_and_src_mask_bool(self):
             src_key_padding_mask=padding_mask,
         )
 
+    def test_decoder_padding_and_src_mask_bool(self):
```
What is this testing? We can use `self.assertEqual` and such to compare outputs.

cpuhrsch replied:
[`assertNoLogs](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertNoLogs) might be able to turn this into a test that fails if we do start to log again.

mikekgfb replied:
Only introduced in 3.11 :(

mikekgfb commented:
@pytorchbot merge

Commit:
Align mask formatting of both masks more closely (#96286)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/96286

Align mask formatting of both masks more closely

Test Plan: sandcastle & github

Differential Revision: D43878634

fbshipit-source-id: 98fd3b1c71c11db22de434445a88ff8cea613ec7

mikekgfb commented:
@pytorchbot merge

mikekgfb commented:
@pytorchbot merge

mikekgfb commented:
@pytorchbot merge

atalman commented:
cc @mikekgfb 

This issue is in the  milestones :[ https://github.com/pytorch/pytorch/milestone/36?closed=1](https://github.com/pytorch/pytorch/milestone/36?closed=1), if you want to see your fix included in this minor release. Please post it as a cherry-pick into the [[v2.0.1] Release Tracker](https://github.com/pytorch/pytorch/issues/97272). 

**The deadline is April 14, 5PM PST.**

**Only issues that have ‘cherry-picks’ will be considered for the release.**

Common FAQs:

Q1: Where can I find more information on the release process and terminology?

A: [pytorch/RELEASE.md at master · pytorch/pytorch · GitHub](https://github.com/pytorch/pytorch/blob/master/RELEASE.md)

Q2: Am I guaranteed to be included in the cherry-pick if I do above?

A: No, it is not guaranteed, the Release Team will review all submissions against the listed criteria before making the final decision on what to include on 4/17.

Q3: When is 2.1 going to be released?

A: We do not have a formal date at this time but will update the community when we do. Our immediate focus is 2.0.1. Note that 1.12 was released on 6/28/22, 1.13 on 10/28/22 and 2.0 on 3/15/23.

Q4: **I missed the 4/14 5PM PST deadline, is there any option to have an extension?**

A: **No, in order to meet our 4/28 goal, we must hold 4/14 as our deadline and will not accept any requests after the fact. We are over communicating the timelines and process with the community to avoid such issues.**

Q5: Where should I double check to see if my issue is in the cherry pick tracker?

A: [[v2.0.1] Release Tracker · Issue #97272 · pytorch/pytorch · GitHub](https://github.com/pytorch/pytorch/issues/97272)

Q6: Where can I find the Release Compatibility Matrix for PyTorch?

A: [pytorch/RELEASE.md at master · pytorch/pytorch · GitHub](https://github.com/pytorch/pytorch/blob/master/RELEASE.md#release-compatibility-matrix)

Please contact OSS Releng team members if you have any questions/comments. Again we appreciate everyone’s time and commitment to the community, PyTorch and 2.0 and 2.01 releases!


Please refer to this post for more details: https://dev-discuss.pytorch.org/t/pytorch-release-2-0-1-important-information/1176
